custom-configurations/src/main/java/com/sequenceiq/cloudbreak/repository/CustomConfigsRepository.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.cloudbreak.repository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+import com.sequenceiq.authorization.service.model.projection.ResourceCrnAndNameView;
+import com.sequenceiq.cloudbreak.domain.CustomConfigs;
+import com.sequenceiq.cloudbreak.workspace.repository.EntityType;
+import org.springframework.data.repository.query.Param;
+
+@EntityType(entityClass = CustomConfigs.class)
+@Transactional(Transactional.TxType.REQUIRED)
+public interface CustomConfigsRepository extends JpaRepository<CustomConfigs, Long> {
+
+    @Query("SELECT c FROM CustomConfigs c LEFT JOIN FETCH c.configurations WHERE c.name = :customConfigsName AND c.account = :accountId")
+    Optional<CustomConfigs> findByNameAndAccountId(@Param("customConfigsName") String customConfigsName, @Param("accountId") String accountId);
+
+    @Query("SELECT c FROM CustomConfigs c LEFT JOIN FETCH c.configurations WHERE c.crn = :resourceCrn")
+    CustomConfigs findByCrn(@Param("resourceCrn") String resourceCrn);
```

Comment:
`find` should return an optional while `get` would fail if there is no result. It's working the way you are using it, but it's a convention. see: https://tuhrig.de/find-vs-get/

Reply:
Acknowledged, will update the return type according to the convention.